components/prism-dax.js

```diff
@@ -0,0 +1,27 @@
+Prism.languages.dax = {
+    'comment': {
+        pattern: /(^|[^\\])(?:\/\*[\s\S]*?\*\/|(?:--|\/\/).*)/,
+        lookbehind: true
+    },
+    'data-field': {
+        pattern: /'[\w '"]+'(?:\[(?:[^\]])+\](?!\])){0,1}|\w+\[(?:[^\]])+\](?!\])/,
```

Comment:
From my understanding, there can be basically any character in between the single quotes or square brackets, right? I'm kinda shocked that it allows single quotes inside single quotes cause it's just... weird.

Do field and table names allow other non-Latin characters (e.g. Japanese like こんにちは)?

Also, what's the reason behind the `(?!\])`?

(And please use `?` instead of `{0,1}`)

Reply:
I was also surprised, you can have table names with `'` or `"` on them. Also Unicode character can be in field or table names. I have highlighted them as an example below. What is the best way to change regex pattern for handling Unicode characters?
![image](https://user-images.githubusercontent.com/7863452/76529405-87771880-6472-11ea-96fc-0881312f91d0.png)

Reply:
In your example, it seems like single-quoted table names behave like strings (you even have to do the `''` to escape one `'`) and the [spec reads like this as well](https://docs.microsoft.com/en-us/dax/dax-syntax-reference#tables).

For the column names, we can use `[ \w\xA0-\uFFFF]` to approximate all Unicode letters plus spaces. So for column names, we get `\[[ \w\xA0-\uFFFF]+\]`. (Should this be copied to `measure`?)

```suggestion
        pattern: /'(?:[^']|'')*'(?!')(?:\[[ \w\xA0-\uFFFF]+\])?|\w+\[[ \w\xA0-\uFFFF]+\]/,
```

Reply:
Thanks for this help. I have made the changes, updated test cases